clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -14,6 +14,7 @@ let Component = "Sema" in {
 let CategoryName = "Semantic Issue" in {
 def note_previous_decl : Note<"%0 declared here">;
 def note_entity_declared_at : Note<"%0 declared here">;
+def note_nullptr_used : Note<"nullptr is a prvalue of type std::nullptr_t">;
```

Comment:
I don't see a point in adding this diagnostic. nullptr is a prvalue of type std::nullptr_t is a known fact for C++ developers.
I think if you want to make diagnostic more clear to show what is wrong with the kernel name especially in case of typedefs we can make a note like: `note: 'the_type_used' aka 'std::some_type' is used here` .

Reply:
Is this the final say on this? The above note was added based on @elizabethandrews 's opinion. 
There's been enough time spent on this than it warrants and I would like to have one final confirmation before anyone else has a different opinion.

Reply:
As I had mentioned earlier, I personally do not think a note is required, but if you disagree I am not opposed to it. What I meant by my earlier comment is - IF you disagree and want to add the note, I would prefer it a separate diagnostic instead of the long diagnostic (stated here - https://github.com/intel/llvm/pull/2644#issuecomment-716768895). Based on feedback from @Fznamznon and @premanandrao, if you are adding the additional note, the content needs to change. If this is outside of the purview of this patch, don't add the additional note

Reply:
Think you are missing my point here - I was pointing to the fact that I took your suggestion for the wordings of the note.
See your comment [here](https://github.com/intel/llvm/pull/2644#issuecomment-716773707)
This is different from @Fznamznon's preference of the note's wordings.
I was just wanting to have one final confirmation on what exactly the note needs to say.

Reply:
If the note were to add additional information such as the where the type got aliased to (my suggestion [here ](https://github.com/intel/llvm/pull/2644#issuecomment-716782258)) or what the typedef/alias is (@Fznamznon's suggestion [here](https://github.com/intel/llvm/pull/2644#discussion_r512505643)) that might be helpful.   If you want to add that as a separate patch that is fine by me.  In that case, don't add any note here.

Reply:
> If the note were to add additional information such as the where the type got aliased to (my suggestion [here ](https://github.com/intel/llvm/pull/2644#issuecomment-716782258)) or what the typedef/alias is (@Fznamznon's suggestion [here](https://github.com/intel/llvm/pull/2644#discussion_r512505643)) that might be helpful

Alright, I can do that. I was given 3 different feedbacks from each of you on how the note should look like. 
I just wanted us to agree upon one version so we don't keep going back and forth.
I can try both @premanandrao's suggestion as well as @Fznamznon's suggestion and see which one I can implement. That will be part of a separate patch though, and the note in this patch will be reverted!

Reply:
I think we need to stop arguing and move note adding to a separate patch to keep this one focused and NFC.
I personally don't have a strong preference regarding the note text (except that it shouldn't point to the known C++ facts), I don't mind if we pick up @premanandrao's suggestion or don't add any notes at all.

Reply:
I wouldn't label it as an argument, this was a discussion on what the next steps should be, which I have concluded in my previous comment.

Reply:
I only meant that I was agree with a part of your last conclusion:
> That will be part of a separate patch though, and the note in this patch will be reverted!

and we should move discussion about the note to a separate issue/PR.